### DISCUSSION TITLE
Variable Precedence in Bruno

### DISCUSSION THREAD

**Discussion by helloanoop** [usebruno/bruno#2332] [Ideas]
Variable Precendence in Bruno

The order of precedence of variables in Bruno when executing a request is as follows:
1. Global Variables - Yet to Be Implemented
2. Collection Variables - Yet To Be Implemented
3. Environment Variables - Already Implemented
4. Secret Variables - Yet to Be Implemented (From Vault, AWS Secrets Manager, etc.)
5. Folder Variables - Yet To Be Implemented
6. Request Variables - Already Implemented
7. Runtime Variables - Already Implemented - `getVar()`, `setVar()`

In case of a conflict, the variable with the highest precedence will be used. For example, if a variable is defined at the request level and at the environment level, the request level variable will be used.

Just adding the precendence order.
Will document more details later on each variable type

**Comment by benjamin-uiris** [comment]
Now that everything is implemented, it would be nice to update the documentation
- Global is missing
- precedence doesn't really apply to secrets since they have their own way to be accessed, right ?

https://docs.usebruno.com/get-started/variables/overview
